racket/collects/racket/hash.rkt

```diff
@@ -63,6 +63,24 @@
                     (combine/key k v (hash-ref hm k))))
         res)))
 
+(define (hash-filter ht pred)
+  (cond
+    [(immutable? ht)
+     (for/fold ([ht ht]) ([(k v) (in-immutable-hash ht)]
+                          #:when (not (pred k v)))
```

Comment:
There's `#:unless`, FWIW. But maybe you want to keep `#:when` to be consistent with `#:when` below, which is fine too.

Reply:
sure thing, changed to unless